src/os/resourcesdir.zig

```diff
@@ -1,6 +1,9 @@
 const std = @import("std");
 const builtin = @import("builtin");
 const Allocator = std.mem.Allocator;
+const build_config = @import("../build_config.zig");
+const internal_os = @import("main.zig");
+const glib = @import("glib");
```

Comment:
The Zig GObject bindings shouldn't be used outside of `src/gtk`. _Maybe_ we could get away with it inside the `comptime` block lower down, but this seems more like something that should perhaps be handled by each individual apprt, which requires a larger discussion.

Reply:
I agree that this should be moved inside apprt-specific code.

Reply:
This code ends up being used by libghostty so I agree that we shouldn't have a hard requirement on glib for libghostty if we can help it. 

That being said, libghostty does expect to find a resources dir without any apprt loaded, so it can't be GTK-specific either (any app can presumably use libghostty without GTK). 

I suggest we allow apprt's to export a function with custom resourcesDir logic that we run first if it exists. This has to happen globally very early in libghostty's initialization so I suggest perhaps: `apprt.runtime.resourcesDir` with the same signature as this one.

Reply:
A key file is really just an ini file, so an alternative would be to use a zig-based parser if you think that'd work better.

Reply:
Adding a INI parser seems like more work than is necessary. Adding `apprt.runtime.resourcesDir` in the way that Mitchell suggests shouldn't be very difficult depending on your experience with Zig. Let us know if you'd like some more pointers.

Reply:
I found an issue with the current approach: The resource path returned is accessible by applications running on the host (e.g. spawned shells), but Ghostty itself cannot access them and must use `/app`.

A solution I'm having in mind is to add a new function to `os.flatpak` to host-ify flatpak paths, exclusively used for `termio.Exec.Subprocess` environment setup, but I'm not sure if this is the way to roll.